src/VisualStudio/Core/Def/Packaging/PackageInstallerServiceFactory.cs

```diff
@@ -361,42 +391,56 @@ private void OnSourceProviderSourcesChanged(object sender, EventArgs e)
             }
         }
 
+        private async Task UpdateStatusBarAsync(EnvDTE.DTE dte, string text, CancellationToken cancellationToken)
+        {
+            await this.ThreadingContext.JoinableTaskFactory.SwitchToMainThreadAsync(cancellationToken);
+            dte.StatusBar.Text = text;
+        }
+
         private static string GetStatusBarText(string packageName, string? installedVersion)
             => installedVersion == null ? packageName : $"{packageName} - {installedVersion}";
 
-        private bool TryUninstallPackage(
-            string packageName, Guid projectGuid, EnvDTE.DTE dte, EnvDTE.Project dteProject)
+        private async Task<bool> TryUninstallPackageAsync(
+            string packageName, Guid projectGuid, EnvDTE.DTE dte, EnvDTE.Project dteProject,
+            IProgressTracker progressTracker, CancellationToken cancellationToken)
         {
-            this.AssertIsForeground();
             Contract.ThrowIfFalse(IsEnabled);
 
-            // TODO: consider putting this under a TWD so it would actually be cancellable.
-            var cancellationToken = CancellationToken.None;
+            // explicitly switch to BG thread to do the installation as nuget installer APIs are free threaded.
+            await TaskScheduler.Default;
```

Comment:
💡 This is going to switch to the background thread and then _immediately_ switch back to the main thread to update the status bar. This line should be moved down to right before the `try`.

Reply:
seems reasonable.  but in that case, why do we need the switch at all?  the UpdateStatusBarAsync call has CA(false) on it.  So we'll be off hte UI thread then, right?